src/psfmachine/ffi.py

```diff
@@ -580,20 +613,63 @@ def _mask_pixels(self, pixel_saturation_limit=1.2e5, magnitude_bright_limit=8):
         self.non_sat_pixel_mask = ~self._saturated_pixels_mask(
             saturation_limit=pixel_saturation_limit
         )
+        # tolerance dependens on pixel scale, TESS pixels are 5 times larger than TESS
+        tolerance = 5 if self.meta["MISSION"] == "TESS" else 25
         self.non_bright_source_mask = ~self._bright_sources_mask(
-            magnitude_limit=magnitude_bright_limit
+            magnitude_limit=magnitude_bright_limit, tolerance=tolerance
         )
-        good_pixels = self.non_sat_pixel_mask & self.non_bright_source_mask
+        self.pixel_mask = self.non_sat_pixel_mask & self.non_bright_source_mask
 
-        self.column = self.column[good_pixels]
-        self.row = self.row[good_pixels]
-        self.ra = self.ra[good_pixels]
-        self.dec = self.dec[good_pixels]
-        self.flux = self.flux[:, good_pixels]
-        self.flux_err = self.flux_err[:, good_pixels]
+        if not hasattr(self, "source_mask"):
+            self._get_source_mask()
+            # include saturated pixels in the source mask and uncontaminated mask
+            self._remove_bad_pixels_from_source_mask()
 
         return
 
+    def _get_source_mask(
+        self,
+        upper_radius_limit=28.0,
+        lower_radius_limit=4.5,
+        upper_flux_limit=2e5,
+        lower_flux_limit=100,
+        correct_centroid_offset=True,
+        plot=False,
+    ):
+        """
+        Adapted version of `machine._get_source_mask()` that masks out saturated and
+        bright halo pixels in FFIs.
+        """
+        super()._get_source_mask(
+            upper_radius_limit=upper_radius_limit,
+            lower_radius_limit=lower_radius_limit,
+            upper_flux_limit=upper_flux_limit,
+            lower_flux_limit=lower_flux_limit,
+            correct_centroid_offset=correct_centroid_offset,
+            plot=plot,
+        )
+        self._remove_bad_pixels_from_source_mask()
+
+    def build_shape_model(
+        self, plot=False, flux_cut_off=1, frame_index="mean", bin_data=False, **kwargs
+    ):
+        """
+        Adapted version of `machine.build_shape_model()` that masks out saturated and
+        bright halo pixels in FFIs.
+        """
+        # call method from super calss `machine`
+        super().build_shape_model(
+            plot=False,
+            flux_cut_off=flux_cut_off,
+            frame_index=frame_index,
+            bin_data=bin_data,
+            **kwargs,
+        )
+        # include sat/halo pixels again into source_mask
+        self._remove_bad_pixels_from_source_mask()
+        if plot:
+            return self.plot_shape_model(frame_index=frame_index, bin_data=bin_data)
+
```

Comment:
These are necessary because in FFI we are including saturated and halo pixels in the `source_mask` and `uncontaminated_pixel_mask`. Then if we want to rebuild the source mask using `machine._get_source_mask()` with different params, we need to include the bad pixels again. Same applies for `machine.build_shape_model()` because the method computes a new source_mask in each call in order to account for possible new `flux_cut_off` value (controls how deep in the background the mask goes), we need to add the bad pixels to the new `source_mask`.